### PR TITLE
WIP - use dynamic last update badge to update at real-time

### PR DESCRIPTION
**Pull request by rluvaton**
The current last update badge need to be updated manually, so I used dynamic badge that fetch from custom endpoint to get the last commit date in master

The code for the custom endpoint is [here](https://runkit.com/rluvaton/get-last-contribution-to-github-branch)

### CODE REVIEW

**Comment by goldbergyoni** [comment]
@rluvaton Amazing idea man, kudos. Also appreciate the code, short functions, very clean.

One concern - I don't want to couple our page to 3rd party services that we don't control, any thoughts on how to the service part of this repo?

**Comment by rluvaton** [comment]
> @rluvaton Amazing idea man, kudos. Also appreciate the code, short functions, very clean.

Thanks, man, appreciate it!

> One concern - I don't want to couple our page to 3rd party services that we don't control, any thoughts on how to the service part of this repo?

I understand your concern, If you want, shields.io provide this functionality but without the day, only the month and a year

GitHub last commit (branch)

```html
<img alt="📅 Last Update" src="https://img.shields.io/github/last-commit/goldbergyoni/nodebestpractices/master?label=%F0%9F%93%85%20Last%20update&logoColor=green">
```


(Because you updated this repo today I'm attaching a badge example of my (first) project that haven't been updated in a long time)
📅 Last Update

(The color should be green, I don't know why it's red (maybe because it was a long time ago? 🤔)

**Comment by josh-hemphill** [comment]
Technically you could directly use the GitHub api with shields.io and you can narrow down the commits by ones effecting a specific file in the repo, though it doesn't format the date.


```md
<img src="https://img.shields.io/badge/dynamic/json?color=green&label=%F0%9F%93%85%20Last%20Updated&query=%24%5B%3A1%5D.commit.author.date&url=https%3A%2F%2Fapi.github.com%2Frepos%2Fgoldbergyoni%2Fnodebestpractices%2Fcommits" />
```

Another option would be to add a github action that updates it on commits.

**Comment by rluvaton** [comment]
> Another option would be to add a github action that updates it on commits.

Thanks man for jumping in! Great idea, this is actually the best way I thin, because:
1. Only update on actual change
2. Control the format at any time
3.With 3rd party API we can reach the GitHub rate limit if the api get abused (if we don't provide proper handling like I did in my code)

File: .github/update-date-in-last-update-badge.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Update date in last update badge
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  run:
+    name: Update the date in last update badge to today
+    runs-on: ubuntu-latest
```
Comment (goldbergyoni) [inline .github/update-date-in-last-update-badge.yml:10]: Nitty-gritty: Security-wise, prefer specific image version

**Review by goldbergyoni — APPROVED** [review]
@rluvaton Awesome work! I skimmed through the bash script *briefly*, trust that you test it and will follow over the next few days that it works?

💜

@josh-hemphill Great insights as always

File: .github/update-date-in-last-update-badge.yml (hunk 1 above)
Comment (rluvaton) [reply .github/update-date-in-last-update-badge.yml:10]: Absolutely right

**Comment by rluvaton** [comment]
> @rluvaton Awesome work! I skimmed through the bash script *briefly*, trust that you test it and will follow over the next few days that it works?


@goldbergyoni I tested the bash script locally but the PR wasn't ready to be merged (this is why I wrote WIP in the PR title), it doesn't create a PR with the changes. But since you merged it I see that the GitHub Action wasn't triggered at all

**Pull request merged**